### ISSUE TITLE
FTP: The remote server returned an error: (530) Not logged in.

### ISSUE THREAD

**Issue by rajeshyalamarthi** [microsoft/BotBuilder-Samples#2508]
Hi, I have stuck with some weird scenarios. 
What I'm trying to do is Fetching the files from one azure web app service[.netcore3.1] and displaying those files in another app service [.netcore3.1] using FTP.
Code I'm using

```
string ftp = _appConfiguration["FTPCredentials:FTPURL"];
                string ftpFolder = _appConfiguration["FTPCredentials:ftpFolder"];
                FtpWebRequest request = (FtpWebRequest)WebRequest.Create(new Uri(ftp+ftpFolder));
                request.Credentials = new NetworkCredential(_appConfiguration["FTPCredentials:UserId"], _appConfiguration["FTPCredentials:Password"]);
                request.Method = WebRequestMethods.Ftp.ListDirectory;
                request.UsePassive = true;
                request.UseBinary = true;
                request.EnableSsl = false;
                FtpWebResponse response = (FtpWebResponse)request.GetResponse();
                StreamReader streamReader = new StreamReader(response.GetResponseStream());
                string line = streamReader.ReadLine();
                while (!string.IsNullOrEmpty(line))
                {
                    Folderlist.Add(line);
                    line = streamReader.ReadLine();
                }

                streamReader.Close();
```

Issues I'm facing:
* When I'm trying to run the code locally where I'm trying to fetch and display the files from azure web app it is working absolutely fine. But when I'm trying to publish the code in azure from there I'm not getting any data(files) from FTP[Tested by keeping logs]. Most of the time it is not getting any data and some times it is returning error like "The remote server returned an error(530) not logged in".

* And The other scenario is my code is in .netcore 2.2 And I have my previous azure app services(.netcore 2.2) in the different azure subscription. If I'm trying to deploy my code in my previous app services then it is working pretty fine. And also I have compared the properties of the app services which are same other than .netcore version.

* And I also tested in vice-versa scenario for the above case. If I'm trying to access the FTP of my previous web app service in my current app service it is getting the data and displaying the files.

I'm completely locked out on this.

**Comment by dmvtech** [comment]
This issue looks to be related to the Azure infrastructure. This doesn't seem to be independent of the bot framework.

Please open a support request through the Azure portal to further troubleshoot this issue.

**Comment by maciakt** [comment]
I was having problems in general connecting to a FTP server with C# code and getting an error like FTPWebRequest 530 Error: Not Logged in.

Turns out, the ftp site I was trying to connect to uses FTP with TLS as noted by this when I could connect to it with Filezilla (proving my credentials username and pw were valid):

Status:	Connection established, waiting for welcome message...
Status:	Initializing TLS...
Status:	Verifying certificate...
Status:	TLS connection established.
Status:	Logged in

I just had to add to my FtpWebRequest to have request.EnableSSL = true; and then it worked and no longer received the 530 error...  example:


	FtpWebRequest request = (FtpWebRequest)WebRequest.Create(serverUri);
    request.Method = WebRequestMethods.Ftp.ListDirectory;
    request.EnableSsl = true;

Hope that helps some people out there!

TJ
